### PR TITLE
DE-31054 Add like symbol operator

### PR DESCRIPTION
**Pull request by vharnach**

### CODE REVIEW

File: src/Operator/Like/LikeQueryLanguageOperatorTest.php [hunk 1]
```diff
@@ -33,11 +33,11 @@ public function testOperatorCanBeParsed(string $query): void
...
         return [
-            ['query' => 'brand ~ bmw'],
-            ['query' => 'brand~bmw'],
-            ['query' => 'brand ~bmw'],
-            ['query' => 'brand~ bmw'],
-            ['query' => '  brand     ~          bmw    '],
+            ['query' => 'brand LIKE bmw'],
+            ['query' => 'brandLIKEbmw'],
+            ['query' => 'brand LIKEbmw'],
+            ['query' => 'brandLIKE bmw'],
```
Comment (michal-tichy-84) [inline src/Operator/Like/LikeQueryLanguageOperatorTest.php:39]: [M] I am pretty sure we do not want to support this

**Review by michal-tichy-84 — CHANGES_REQUESTED** [review]

File: src/Operator/Like/LikeQueryLanguageOperatorTest.php (hunk 1 above)
Comment (vharnach) [reply src/Operator/Like/LikeQueryLanguageOperatorTest.php:39]: True, I should have used QueryLanguageOperatorParserCreator::createWordOperatorParser for "word operator"

File: src/Operator/Like/LikeQueryLanguageOperator.php [hunk 2]
```diff
@@ -29,7 +29,7 @@ public function getOperatorIdentifier(): string
      */
     public function createOperatorParser(): MonoParser
     {
-        return QueryLanguageOperatorParserCreator::createSignOperatorParser('~');
```
[the remark this answers is not in the document]
Comment (vharnach) [reply src/Operator/Like/LikeQueryLanguageOperator.php:32]: I used the former class for LIKE operator and Like**Symbol**QueryLanguageOperator class for ~

File: src/Operator/Like/LikeSymbolQueryLanguageOperator.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+use BrandEmbassy\QueryLanguageParser\Operator\QueryLanguageOperatorParserCreator;
+use Ferno\Loco\ConcParser;
+use Ferno\Loco\GrammarException;
+use Ferno\Loco\MonoParser;
+use function assert;
+
+/**
+ * @final
+ */
+class LikeSymbolQueryLanguageOperator implements QueryLanguageOperator
```
[the remark this answers is not in the document]
Comment (vharnach) [reply src/Operator/Like/LikeSymbolQueryLanguageOperator.php:16]: You are right, but to update the whole parser engine to support more than one operator... It does not seem to be worth the effort.

File: src/Operator/Like/LikeQueryLanguageOperatorTest.php (hunk 1 above)
Comment (michal-tichy-84) [reply src/Operator/Like/LikeQueryLanguageOperatorTest.php:39]: So this is a valid scenario?

**Review by michal-tichy-84 — APPROVED** [review]

File: src/Operator/Like/LikeQueryLanguageOperatorTest.php [hunk 4]
```diff
@@ -33,11 +34,33 @@ public function testOperatorCanBeParsed(string $query): void
     public function queryToBeParsedProvider(): array
     {
         return [
-            ['query' => 'brand ~ bmw'],
-            ['query' => 'brand~bmw'],
-            ['query' => 'brand ~bmw'],
-            ['query' => 'brand~ bmw'],
-            ['query' => '  brand     ~          bmw    '],
+            ['query' => 'brand LIKE bmw'],
```
Comment (milan-zelenka) [inline src/Operator/Like/LikeQueryLanguageOperatorTest.php:37]: [M] What about searching for multiple words, e.g. 'bmw m305' ?

**Review by milan-zelenka — APPROVED** [review]

File: src/Operator/Like/LikeQueryLanguageOperatorTest.php (hunk 4 above)
Comment (vharnach) [reply src/Operator/Like/LikeQueryLanguageOperatorTest.php:37]: that is not supported by the query language

**Pull request merged**